hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/cluster/ReplaceCommitValidateUtil.java

```diff
@@ -0,0 +1,49 @@
+package org.apache.hudi.table.action.cluster;
+
+import org.apache.hudi.common.model.HoodieReplaceCommitMetadata;
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.hudi.common.util.JsonUtils;
+import org.apache.hudi.exception.HoodieException;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+
+import static org.apache.hudi.common.util.FileIOUtils.LOG;
+
+public class ReplaceCommitValidateUtil {
+  public static void validateReplaceCommit(HoodieTableMetaClient metaClient) {
+    metaClient.reloadActiveTimeline();
+    Set<String> replaceFileids = new HashSet<>();
+
+    // Verify pending and completed replace commit
+    Stream.concat(metaClient.getActiveTimeline().getCompletedReplaceTimeline().getInstants().stream(),
+        metaClient.getActiveTimeline().filterInflights().getInstants().stream()).parallel().map(instant -> {
+      try {
```

Comment:
Understand the intension that deserialize commit might be costly, but I prefer sequential stream instead of parallel. the parallelism is out of control and usually the merging is costly.

Reply:
I think you are right. I will remove the parallelism. And "the merging is costly" refers to the concat operation? I do not quite understand.

Reply:
Parallel stream requires a merge operation, and usually it could be costly.

Reply:
Thanks for your opinion, I understand now！